sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala

```diff
@@ -174,7 +174,8 @@ case class In(attribute: String, values: Array[Any]) extends Filter {
     case _ => false
   }
   override def toString: String = {
-    s"In($attribute, [${values.mkString(",")}])"
+    // Sort elements for deterministic behaviours
+    s"In($attribute, [${values.map(_.toString).sorted.mkString(",")}])"
```

Comment:
`_.toString` is not null safe, while `Seq(null).mkString` is fine. I think we should use `String.valueOf`